<launch>
    <!-- Launch the driver for the ps4 controller and publish data to "ps4_input" topic -->
    
    <arg name="robot_names"  default='["virtual_leader","mur620a","mur620c","mur620d"]'/>
    <arg name="cmd_vel_topic_prefix"  default=""/>
    <arg name="twist_stamped"  default="False"/>
    
    <!--
    <node name="ps4_controller_driver" pkg="joy" type="joy_node">
        <remap from="joy" to="ps4_input"/>
        <param name="~autorepeat_rate" value="0.0"/>
    </node>-->

    <!-- Get proper velocities from the raw controller data.-->   
    <node name="teleoperate" pkg="ps4_controller" type="PlayStationDiffDrive.py" output="screen">
        <rosparam command="load" file="$(find ps4_controller)/config/controller_config.yaml"/>
        <!-- remap from="joy" to="ps4_input"/ -->  <!--Get data from the "ps4_input" topic-->
        <param name="cmd_vel_topic_prefix" value="$(arg cmd_vel_topic_prefix)" />
        <param name="~autorepeat_rate" value="0.0"/>
        <rosparam param="robot_names" subst_value="True">$(arg robot_names)</rosparam>
        <param name="twist_stamped" value="$(arg twist_stamped)" />
    </node>
</launch>